lib/resources/oidc.js

```diff
@@ -118,13 +118,47 @@ module.exports = (service, endpoint) => {
       // Instead, we need to render a page and then "browse" from that page to the normal frontend:
 
       // id=cl only set for playwright. Why can't it locate this anchor in any other way?
-      return {
-        head: html`<meta http-equiv="refresh" content="0; url=${nextPath}">`,
-        body: html`
-          <h1>Authentication Successful</h1>
-          <div><a href="${nextPath}" id="cl">Continue to ODK Central</a></div>
-        `,
-      };
+      return html`
+        <!doctype html>
+        <html>
+          <head>
+            <title>Loading... | ODK Central</title>
+            <meta http-equiv="refresh" content="0; url=${nextPath}">
```

Comment:
Looking at the `${nextPath}` interpolation, is the `content` attribute being escaped for HTML? Is there any risk of a maliciously constructed `next` query parameter that includes a quote, thereby escaping the attribute? If so, I'd start worrying about XSS and such, e.g., somehow breaking out of the `<meta>` tag and inserting a `<script>` tag. Or does `safeNextPathFrom()` ensure against that? It'd be great if we had tests related to that. (Maybe we do, and I've just missed them.)

Reply:
@matthew-white Great questions, and I think they apply equally to `master` as this PR.

Reply:
@matthew-white PR at https://github.com/getodk/central-backend/pull/1003 for more test cases for `safeNextPathFrom()`

Reply:
Awesome, I just approved that PR. I think there are still issues here though related to ampersands. It looks like `new URL()` doesn't encode `&` and `;`, thereby opening the door to HTML entities. I'm pretty sure that HTML entities are valid in HTML attributes.

For example, `safeNextPathFrom()` prevents `next=/login`, but I'm wondering if HTML entities could be used to bypass that check along the lines of `next=/login/foo&sol;&period;&period;`. As a side note, I actually don't think it's important for `safeNextPathFrom()` to prevent `next=/login`: it's important in Frontend just to prevent the `AccountLogin` component from navigating to itself, which would cause a Vue Router error. But I do think that ampersands open up possibilities that we haven't thought through.

Is there a function we could call to encode `nextPath` for HTML? Could we do that rather than trying to prevent HTML special characters in `nextPath`? I feel like there should be something in Express or another package to do that. I think that would result in a neater separation of concerns, making it clear that `safeNextPathFrom()` is mostly meant to prevent a redirect away from Central, not prevent XSS.

Reply:
> Is there a function we could call to encode `nextPath` for HTML?

I think that's what `safeNextPath()` should be.  If there is a standard approach to this (especially with a path to supporting search/fragment switching & enketo URLs), I'd be happy to use that instead.

> For example, `safeNextPathFrom()` prevents `next=/login`

I'll remove this exception.

> but I'm wondering if HTML entities could be used to bypass that check along the lines of `next=/login/foo&sol;&period;&period;`.

They can.  `next=/login/foo&sol;&period;&period;` will be treated the same as `next=/login/foo/..`, which there isn't a specific problem with.  Perhaps an issue with escaping enketo URLs?  I'll add test cases for that as well.